cli.js

```diff
@@ -2,21 +2,42 @@
 import * as fs from 'fs';
 import * as module from 'module';
 import * as path from 'path';
+import { getopt } from 'stdio';
 
 import puppeteer from 'puppeteer'
 import { createServer } from 'vite'
 
 import { MochaProtocolPlayer } from './mochaProtocolPlayer.js';
 import { MochaProtocolReporter } from './mochaProtocolReporter.js';
 
-// Note: eventually turn into args with default values.
-const port = 3001;
+const options = getopt({
+  port: { key: 'p', description: 'Port for the tests to run on', args: 1, default: '3001', required: false },
+  entry: { key: 'e', description: 'Entry html file that contains Mocha Setup. Relative to CWD', default: 'test.html', args: 1, required: false },
+  reporter: { key: 'r', description: 'Reporter to use. Available: [dot, json, json-stream, list, spec, tap]', default: 'spec', args: 1, required: false },
+  reporterOptions: { key: 'o', description: 'options to pass to the reporter', default: undefined, args: 1, required: false },
+  verbose: { key: 'v', description: 'Verbose Output', default: false, required: false},
+  debug: { key: 'd', description: 'Enable debug mode. Note: test will run until quit via console (ctrl+c)', default: false, required: false},
+});
+
+if(options.args) {
+  options.reporter = options.args[0];
```

Comment:
I am not sure I understand why we need these? I don't think we ever supported reading "reporter" from args[0]. Is this needed for the `npx mocha-vite-puppeteer` use case or for the `test:app` use case?

Reply:
There is only one edge-case where options.args would be defined, which is when a user has an npm script setup like this:
`"test": "mocha-vite-puppeteer"`  
  
And then calls `npm run test --port 3005`  
Although `npm run test 3005` will result in the same behavior.  

Calling npm in such a manner results in those params being "sent" to the "test" script and not directly to mocha-vite-puppeteer. When they do eventually reach cli.js, the stdio package thinks that those are just misc args and throws them under that args key.   
  
My plan for this was to catch an end-user mistake, but we could just remove this or replace it with a console.warn instead.

Reply:
I've updated this to just be a warning for any user that passes malformed args